staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -128,17 +129,21 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		}
 		trace.Step("Conversion done")
 
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			_, name, _ = scope.Namer.ObjectName(obj)
+		}
+		if *gvk == namespaceGVK {
```

Comment:
```suggestion
		if len(namespace) == 0 && *gvk == namespaceGVK {
```

Reply:
Sounds good, I have committed the change 👍